mlrun/runtimes/pod.py

```diff
@@ -660,6 +661,51 @@ def kube_resource_spec_to_pod_spec(
     )
 
 
+def enrich_resources_with_default_pod_resources(resources: dict = None):
+    resources_types = ["cpu", "memory", "gpu"]
+    resource_requirements = ["requests", "limits"]
+    default_resources = mlconf.default_function_pod_resources.to_dict()
+    if not resources:
+        return default_resources
+
+    _verify_gpu_requests_and_limits(
+        requests_gpu=resources.setdefault("requests", {}).setdefault("gpu", ""),
+        limits_gpu=resources.setdefault("requests", {}).setdefault("gpu", ""),
+    )
+    for resource_requirement in resource_requirements:
+        for resource_type in resources_types:
+            if not resources.setdefault(resource_requirement, {}).setdefault(
+                resource_type, ""
+            ):
+                if resource_type == "gpu" and resource_requirement == "requests":
+                    # set gpu defaults only on limits
+                    continue
+
+                resources[resource_requirement][resource_type] = default_resources[
+                    resource_requirement
+                ][resource_type]
+
+            if resources[resource_requirement][resource_type]:
+                verify_field_regex(
+                    field_name=f"function.spec.resources.{resource_requirement}.{resource_type}",
+                    field_value=resources[resource_requirement][resource_type],
+                    patterns=mlrun.utils.regex.k8s_resource_quantity_regex,
+                )
+    return resources
+
+
+def _verify_gpu_requests_and_limits(requests_gpu: str = None, limits_gpu: str = None):
+    # https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/
+    if requests_gpu and not limits_gpu:
+        raise mlrun.errors.MLRunConflictError(
+            "You cannot specify GPU requests without specifying limits"
+        )
+    if requests_gpu and limits_gpu and requests_gpu == limits_gpu:
+        raise mlrun.errors.MLRunConflictError(
+            "When specifying both GPU requests and limits these two values must be equal"
+        )
```

Comment:
You should probably use this function on the config initialization so if someone configured something that is invalid the API will explode straight away when initializing